src/System.Windows.Forms/src/System/Windows/Forms/TreeNode.cs

```diff
@@ -1812,6 +1813,15 @@ internal TreeView FindTreeView()
             return node.treeView;
         }
 
+        internal void GetChildNodes(List<TreeNode> nodes)
+        {
+            foreach (TreeNode child in Nodes)
+            {
+                nodes.Add(child);
+                child.GetChildNodes(nodes);
+            }
+        }
```

Comment:
A super unobvious method. I would say it is not correct. We shouldn't send a list to "GET" method and change it inside. It's not good. You create a list, send it as an argument and change it inside the method. 
![image](https://user-images.githubusercontent.com/49272759/107214815-88fee700-6a1b-11eb-9bb3-0d6af009620f.png)

I would rework it to:
```suggestion
        internal List<TreeNode> GetChildNodes()
        {
            List<TreeNode> nodes = new List<TreeNode>();

            foreach (TreeNode child in Nodes)
            {
                nodes.Add(child);
                nodes.AddRange(child.GetChildNodes());
            }

            return nodes;
        }
```

Reply:
I have requested this change in https://github.com/dotnet/winforms/pull/4466#discussion_r570066332

Your version will be allocating (possibly a lot of) new lists. With the current implementation we're allocating only once (though we will be resizing).

Reply:
Yes, I agree, we don't need billions of new lists. We decided to rename the method and add a comment to be consistent with common design patterns. In particular, this method should not be "GET"